src/coreclr/jit/jiteh.cpp

```diff
@@ -2127,6 +2127,12 @@ void Compiler::fgNormalizeEH()
 
 #endif // 0
 
+    // For NativeAOT we need to create a fake exception filter for generic handlers
```

Comment:
We should do this everywhere. 

It will simplify testing and it will allow us to delete the patch work that handles this case in the VM: 
https://github.com/dotnet/runtime/blob/4cbe6f99d23e04c56a89251d49de1b0f14000427/src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs#L740-L756
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/coreclr/vm/jitinterface.cpp#L7754-L7794
https://github.com/dotnet/runtime/blob/4cbe6f99d23e04c56a89251d49de1b0f14000427/src/coreclr/vm/codeman.cpp#L3479-L3508
https://github.com/dotnet/runtime/blob/4cbe6f99d23e04c56a89251d49de1b0f14000427/src/coreclr/vm/codeman.cpp#L6004-L6035

Reply:
@jkotas is it OK that it noticeably increases codegen size (for cold code though), e.g. for:
```csharp
[MethodImpl(MethodImplOptions.NoInlining)]
static int Egor<T>(int a, int b) where T : Exception
{
    try
    {
        return a / b;
    }
    catch (T)
    {
        return 22;
    }
}
```
jit-diff is https://www.diffchecker.com/1333mzWq (JIT mode)

Reply:
I think that's fine. The generic exception handlers are rare and they are typically used together with other complex code.